app/controllers/case_contacts/form_controller.rb

```diff
@@ -9,29 +9,21 @@ class CaseContacts::FormController < ApplicationController
 
   # wizard_path
   def show
-    authorize @case_contact
+    manage_form_step
     get_cases_and_contact_types
-    @page = wizard_steps.index(step) + 1
-    @total_pages = steps.count
-
     render_wizard
     wizard_path
   end
 
   def update
-    authorize @case_contact
+    manage_form_step(step_navigation: true)
```

Comment:
Use helper for repeated behavior

Reply:
I'm a little confused what is happening here, seems like a lot of side effects are possible.  also not clear how it interacts with some before_actions, which were already kind of a mess... does it make sense to check for steps in `set_steps` for example?